src/screens/home/Home.tsx

```diff
@@ -1,67 +1,99 @@
-import { View, StyleSheet, Dimensions } from 'react-native';
+import { View, StyleSheet, useWindowDimensions } from 'react-native';
 import MasonryList from 'react-native-masonry-list';
 
 const state = {
   onLoading: true,
   offset: 0,
 };
 
-const screenWidth = Dimensions.get('window').width;
+// [1] Since you're using Functional Components, refrain from using the static
+// `Dimensions.get()` function. Use the `useWindowDimensions` hook instead. 
+// (Refer to the comment section also marked with [1] below.)
 
 const numColumns = 2;
 
-const tileSize = screenWidth / numColumns;
+// [4] What purpose does this global array serves? If it's a temporary variable,
+// define it in the scope. If it is a global variable, one should have a very
+// strong reason to use it. Generally, avoid global singletons as much as you
+// can. Especially those typed with `any`.
+// (Refer to the comment section also marked with [4] below.)
+// let tmp: any[] = [];
 
-const styles = StyleSheet.create({
-  view: {
-    flex: 1,
-    alignItems: 'center',
-    padding: 10,
-  },
-  textInput: {
-    margin: 10,
-    width: '100%',
-    height: 50,
-    color: 'white',
-  },
-  image: {
-    width: tileSize,
-    height: 150,
-    marginBottom: 5,
-  },
-  flatList: {
-    width: screenWidth,
-  },
-});
+// [2] We should define a type for the GIFs instead of using `any`. In our code,
+// we try to avoid using `any` as much as we can, because if we use `any`, then
+// it is no better than just using JavaScript (sometimes could be worse because
+// VS Code can still infer the type using IntelliSense). If you are unfamiliar
+// with the type of the object, you may use Postman to send a test query.
+// (Refer to the comment section also marked with [2] below.)
+interface GiphyTrendingResponse {
+  data: Array<{
+    images: {
+      original: {
+        url: string
+      }
+    }
+  }>
+}
+
+interface GiphySource {
+  url: string
+}
 
-let tmp: any[] = [];
+// [6] Constants should live outside of Functional Components. If not it will be
+// redefined for every render loop.
+// API settings
+const limit = 20;
 
 export default () => {
-  const [gifsSrc, setGifsSrc] = useState<any>([]);
+  // [1] Instead of using `Dimensions.get()` here I'm using a React Native hook
+  // to get the screen width. You are now just writing for iOS and Android,
+  // which screen size remains constant most of the time. However, React Native
+  // can now also compile for Windows and macOS, where the user can resize the
+  // window. Screen width is now therefore reactive, not static.
+  const { width: screenWidth } = useWindowDimensions()
+  const tileSize = screenWidth / numColumns
 
-  // API settings
-  const limit = 20;
+  const [gifsSrc, setGifsSrc] = useState<Array<GiphySource>>([]);
+
+  // [3] Refer to comment also marked [3] below
+  const [offset, setOffset] = useState(0);
+
+  // [5] Refer to comment also marked [5] below
+  const [isLoading, setIsLoading] = useState(false)
```

Comment:
You should have a `useState` hook for every one of your states. Avoid using `state.<statename>`